### PR TITLE
Check curl status in busybox bootstrap module

### PR DESCRIPTION
**Pull request by jtriley**
Fixes https://github.com/singularityware/singularity/pull/438#issuecomment-270187567

Changes proposed in this pull request

 - Fix default busybox MirrorURL (it moved recently)
 - Update busybox bootstrap module to check return value of curl and communicate failure to user

@singularityware-admin

### CODE REVIEW

**Comment by vsoch** [comment]
This looks good for testing - once @gmkurtzer is able to merge the initial fixes (and resolve the current bug) we can rebase and get this fix in as well! Thanks for doing this @jtriley!

**Comment by bbockelm** [comment]
@jtriley - can you rebase this branch against current master so the tests work?

File: libexec/bootstrap/modules-v2/build-busybox.sh [hunk 1]
```diff
@@ -59,7 +59,12 @@ echo "root:!:0:0:root:/root:/bin/sh" > "$SINGULARITY_ROOTFS/etc/passwd"
 echo " root:x:0:" > "$SINGULARITY_ROOTFS/etc/group"
 echo "127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4" > "$SINGULARITY_ROOTFS/etc/hosts"
 
-curl "$MIRROR" > "$SINGULARITY_ROOTFS/bin/busybox"
+curl -f "$MIRROR" > "$SINGULARITY_ROOTFS/bin/busybox"
+
+if [ ! $? -eq 0 ]; then
```
Comment (bbockelm) [inline libexec/bootstrap/modules-v2/build-busybox.sh]: Can you switch this to `[ $1 -ne 0 ]` ?  I know it's equivalent, just requires less thinking.

Comment (jtriley) [reply libexec/bootstrap/modules-v2/build-busybox.sh]: Agreed - I meant to do that last night but got distracted by this test failure (from the tests I added in this PR)

**Comment by jtriley** [comment]
@bbockelm @gmkurtzer this should be all set now. I squashed the commits down and tests are passing now.

**Comment by gmkurtzer** [comment]
Looks good to me, thanks!

**Pull request merged**